File: 18. C# ASP.NET MVC Overview - Exercises/Calculator/CalculatorApp/Models/Calculator.cs
using System;

namespace Calculator_CSharp.Models
{
    public class Calculator
    {
        public Calculator()
        {
            this.Result = 0;
        }

        public decimal LeftOperand { get; set; }
        public decimal RightOperand { get; set; }
        public string Operator { get; set; }
        public decimal Result { get; set; }

        public void CalculateResult()
        {
            var resultDouble = 0.00;

            switch (this.Operator)
            {
                case "+": this.Result = this.LeftOperand + this.RightOperand; break;
                case "-": this.Result = this.LeftOperand - this.RightOperand; break;
                case "*": this.Result = this.LeftOperand * this.RightOperand; break;
                case "/": this.Result = this.LeftOperand / this.RightOperand; break;
                case "^": resultDouble = Math.Pow((double)this.LeftOperand, (double)this.RightOperand); this.Result = (decimal)resultDouble; break;
                case "√": resultDouble = Math.Sqrt((double)this.LeftOperand); this.Result = (decimal)resultDouble; break;
                case "Mod": this.Result = this.LeftOperand % this.RightOperand; break;
                default: this.Result = 0; break;
            }
        }
    }
}

File: 07. JavaScript ExpressJS Overview/Calculator/models/Calculator.js
function Calculator (leftOperand, rightOperand, operator) {
    this.leftOperand = leftOperand;
    this.rightOperand = rightOperand;
    this.operator = operator;

    this.calculateResult = function () {
        let result = 0;

        switch (this.operator) {
            case '+':
                result = this.leftOperand + this.rightOperand;
                break;
            case '-':
                result = this.leftOperand - this.rightOperand;
                break;
            case '*':
                result = this.leftOperand * this.rightOperand;
                break;
            case '/':
                result = this.leftOperand / this.rightOperand;
                break;
            case '^':
                result = Math.pow(this.leftOperand, this.rightOperand);                
                break;
            case'√':
                result = Math.sqrt(this.leftOperand);                
            break;
            case'Mod':
                result = this.leftOperand % this.rightOperand;                
            break;
        }
        return result;
    }
}

module.exports = Calculator;